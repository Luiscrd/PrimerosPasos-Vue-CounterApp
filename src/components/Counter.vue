<template>
    <div class="d-flex flex-column">
        <div class="container mt-4">
            <h1>{{ title }}: {{ start }}</h1>
            <hr>
            <div class="d-flex align-items-center">
                <h3>Numero: </h3>
                <button type="button" class="btn btn-primary" @click="decrese()">-1</button>
                <h3>{{ counter }}</h3>
                <button type="button" class="btn btn-primary" @click="icrise()">+1</button>
            </div>
            <hr>
            <h3>Numero al cuadrado : {{ counter }} <sup>2</sup> = {{ squereCounter }}</h3>
            <hr>
        </div>
    </div>
</template>

<script>
export default {
    name: 'Couter',
    props: {
        title: {
            type: String,
            defaul: 'Counter'
        },
        start: {
            type: Number,
            defaul: 10,
            validator( value ) {
                return value > 0
            }
        }
    },
    data() {
        return {
            counter: this.start
        }
    },
    methods: {
        icrise() {
            this.counter++
        },
        decrese() {
            this.counter--
        }
    },
    computed: {
        squereCounter() {
            return this.counter * this.counter
        }
    }
}
</script>

<style scoped>
.btn:hover {
    color: white;
    background-color: #11613e;
    border-color: #11613e;
}

[type=button]:not(:disabled),
[type=reset]:not(:disabled),
[type=submit]:not(:disabled),
button:not(:disabled) {
    cursor: pointer;
}

.btn-primary {
    width: 50px;
    text-align: center;
    margin: 0 10px;
    --bs-btn-color: #fff;
    --bs-btn-bg: #41B783;
    --bs-btn-border-color: #41B783;
    --bs-btn-hover-color: #fff;
    --bs-btn-hover-bg: #41B783;
    --bs-btn-hover-border-color: #41B783;
    --bs-btn-focus-shadow-rgb: 49, 132, 253;
    --bs-btn-active-color: #fff;
    --bs-btn-active-bg: #41B783;
    --bs-btn-active-border-color: #41B783;
    --bs-btn-active-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
    --bs-btn-disabled-color: #fff;
    --bs-btn-disabled-bg: #41B783;
    --bs-btn-disabled-border-color: #41B783;
}

h3 {
    margin: 0 10px;
    display: inline;
}
</style>