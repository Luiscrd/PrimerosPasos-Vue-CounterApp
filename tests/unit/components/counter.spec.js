import { shallowMount } from '@vue/test-utils'
import Counter from '@/components/Counter.vue'

describe('Counter Component', () => {

    let wrapper;

    beforeEach(() => {

        wrapper = shallowMount(Counter)

    });

    test('Debe de hacer mach con el snapshot', () => {

        expect(wrapper.html()).toMatchSnapshot()

    })

    test('Etiqueta h1 debe de tener el valor por defecto "Counter: 10"', () => {

        expect(wrapper.find('h1').exists()).toBeTruthy()

        const h1Value = wrapper.find('h1').text()

        expect(h1Value).toBe('Counter: 10')

    })

    test('El valor por defecto debe ser 10 en el h3', () => {

        const h3 = wrapper.find('[data-testid="counter"]')

        expect(h3.exists()).toBeTruthy()

        expect(h3.text()).toBe('10')

    })

    test('El Valor del counter debe incrementarse y decrementarse', async () => {

        let value = wrapper.find('[data-testid="counter"]').text()

        expect(value).toBe('10')

        const [decreseBtn, incriseBtn] = wrapper.findAll('button')

        await incriseBtn.trigger('click')

        value = wrapper.find('[data-testid="counter"]').text()

        expect(value).toBe('11')

        await decreseBtn.trigger('click')
        await decreseBtn.trigger('click')

        value = wrapper.find('[data-testid="counter"]').text()

        expect(value).toBe('9')

    })

    test('Debe de establecer el valor por defecto', () => {

        const h1Value = wrapper.find('h1').text()

        const title = wrapper.props('title')

        const start = wrapper.props('start')

        expect(h1Value).toBe(`${title}: ${start}`)

    })

    test('Debe de mostrar las propeties enviadas', () => {

        const title = 'Prueba'

        const start = 20

        const wrapper = shallowMount(Counter, {
            props: {
                title,
                start
            }
        })

        const h1Value = wrapper.find('h1').text()

        expect(h1Value).toBe('Prueba: 20')

    })


})