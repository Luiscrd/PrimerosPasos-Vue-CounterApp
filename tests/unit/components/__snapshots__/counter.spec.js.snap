// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Debe de hacer mach con el snapshot 1`] = `
<div class="d-flex flex-column">
  <div class="container mt-4">
    <h1>Counter: 10</h1>
    <hr>
    <div class="buetons-box d-flex align-items-center">
      <h3>Numero: </h3><button type="button" class="btn btn-primary">-1</button>
      <h3 data-testid="counter">10</h3><button type="button" class="btn btn-primary">+1</button>
    </div>
    <hr>
    <h3>Numero al cuadrado : 10 <sup>2</sup> = 100</h3>
    <hr>
  </div>
</div>
`;
